sasl/aws_msk_iam_v2/msk_iam.go

```diff
@@ -32,15 +32,15 @@ const (
 	queryExpiryKey   = "X-Amz-Expires"
 )
 
-var signUserAgent = fmt.Sprintf("kafka-go/sasl/aws_msk_iam/%s", runtime.Version())
+var signUserAgent = fmt.Sprintf("kafka-go/sasl/aws_msk_iam_v2/%s", runtime.Version())
```

Comment:
Very minor and unrelated to your change, but since it's being modified. This could be a simpler way to express the concatenation:
```suggestion
var signUserAgent = "kafka-go/sasl/aws_msk_iam_v2/" + runtime.Version()
```